### PR TITLE
Added Rustdoc support for SDK

### PR DESCRIPTION
**Pull request by dominic-mulligan-arm**
As part of a wider task of getting Rustdoc built for the entire Veracruz codebase, where appropriate, this change adds:

- New Makefile targets for building Rustdocs for the major components of the Veracruz SDK,
- Various fixes to Rustdocs appearing in source files in the SDK,
- Fixes to the Rustdocs appearing in `chihuahua`.

Documentation for the SDK can now be built by typing `make doc` in the top-level SDK Makefile.  Documentation appears under `target/doc/` in each subdirectory of the SDK, e.g. `examples/linear-regression/target/doc/linear-regression/index.html`.

### CODE REVIEW

**Comment by dominic-mulligan-arm** [comment]
This is working towards fixing issue #29.

**Review by basmaelgaabouri — DISMISSED** [review]

File: sdk/libveracruz/Makefile [hunk 1]
```diff
@@ -13,6 +13,9 @@ VERACRUZ_TARGET_PATH = $(shell cd ../ && pwd)
 all:
 	XARGO_RUST_SRC=$(RUST_STDLIB_SRC_DIR) RUST_TARGET_PATH=$(VERACRUZ_TARGET_PATH) xargo build --target wasm32-arm-veracruz --release
 
+doc:
```
Comment (ShaleXIONG) [inline sdk/libveracruz/Makefile:16]: This file does not have a license declaration and .PHONY declaration. Can you add it.

**Comment by veracruz-project-owner** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: Veracruz
* Commit ID: 77ec0cbd7dce386dfad79e589c5ca1ec6140c438
* Result: SUCCEEDED
* [Build Logs](https://t4r5bdzqe9.execute-api.eu-west-2.amazonaws.com/Prod/buildlogs?key=d3099cd6-4c14-447f-af90-9ac42b838748%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Comment by veracruz-project-owner** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: Veracruz
* Commit ID: 9972761ac5c72aa5db10349543c3509e99e5bfb7
* Result: FAILED
* [Build Logs](https://t4r5bdzqe9.execute-api.eu-west-2.amazonaws.com/Prod/buildlogs?key=1f265712-48be-43f5-b88c-547c81ac9ba8%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Comment by veracruz-project-owner** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: Veracruz
* Commit ID: 3c8781dda8ea80d0f97265dba20126e73c58fe70
* Result: SUCCEEDED
* [Build Logs](https://t4r5bdzqe9.execute-api.eu-west-2.amazonaws.com/Prod/buildlogs?key=9c3e68af-6964-412d-910a-19eea571e2d7%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Review by ShaleXIONG — APPROVED** [review]

**Review by ShaleXIONG — APPROVED** [review]

**Comment by veracruz-project-owner** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: Veracruz
* Commit ID: e865367218a838dfaf1f53fe99717bfa208458d3
* Result: SUCCEEDED
* [Build Logs](https://t4r5bdzqe9.execute-api.eu-west-2.amazonaws.com/Prod/buildlogs?key=0cf2a866-2655-4d31-9190-943bcc540e04%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Review by basmaelgaabouri — APPROVED** [review]

**Comment by dominic-mulligan-arm** [comment]
+2 reached, merging.

**Pull request merged**